cmd/thanos/rule.go

```diff
@@ -200,6 +201,55 @@ func registerRule(m map[string]setupFunc, app *kingpin.Application) {
 	}
 }
 
+// RuleMetrics defines thanos rule metrics.
+type RuleMetrics struct {
+	configSuccess     prometheus.Gauge
+	configSuccessTime prometheus.Gauge
+	duplicatedQuery   prometheus.Counter
+	rulesLoaded       *prometheus.GaugeVec
+	ruleEvalWarnings  *prometheus.CounterVec
+}
+
+func newRuleMetrics(reg *prometheus.Registry) *RuleMetrics {
+	m := new(RuleMetrics)
+
+	m.configSuccess = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Do you want to be our super hero (level: legendary)? Let's start using new pattern here as decided here: https://github.com/thanos-io/thanos/issues/2102

so `promauto.With(reg).NewGauge... and all here :hugs:

Reply:
Checked the promauto package. The promauto.With function exists in master branch only. There may be something unstable there. Not sure we should use the master branch.
@bwplotka  What's your suggestion?

Reply:
I maintain master, let's update maybe? (:

Reply:
@bwplotka Cool. Updated, please help check.